### PR TITLE
Trenches - Add modded Entrenching Tools support

### PR DESCRIPTION
**Pull request by veteran29**
**When merged this pull request will:**
- title

### CODE REVIEW

File: addons/trenches/README.md [hunk 1]
```diff
@@ -1,17 +1,7 @@
...
-Single surfaces can be whitelisted by adding `ACE_canDig = 1` into `CfgSurfaces`.
-Example:
-```cpp
-class CfgSurfaces {
-    class myAwesomeSurface {
-        ACE_canDig = 1;
-    };
-};
-```
+Provides players with capability of digging trenches.
```
Comment (rautamiekka) [inline addons/trenches/README.md:4]:
```suggestion
Provides players with the capability of digging trenches.
```

**Review by mharis001 — APPROVED** [review]

**Comment by PabstMirror** [comment]
cup has a few shovel backpacks
but it would require also scanning cfgVics

**Comment by veteran29** [comment]
Will look into adding CfgVehicles support. I think we might have some packs with shovels in SOG too. Neat idea!

File: addons/trenches/XEH_preStart.sqf [hunk 2]
```diff
@@ -2,7 +2,11 @@
 
 #include "XEH_PREP.hpp"
 
+private _entrenchingTools = 
+(QUOTE(getNumber (_x >> QQGVAR(entrenchingTool)) > 0) configClasses (configFile >> "CfgWeapons") apply {configName _x}) +
+(QUOTE(getNumber (_x >> QQGVAR(entrenchingTool)) > 0) configClasses (configFile >> "CfgVehicles") apply {configName _x});
```
Comment (veteran29) [inline addons/trenches/XEH_preStart.sqf:7]: `append` should be faster.

**Review by veteran29 — COMMENTED** [review]
Pabst changes LGMT

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**